sinks/cortex/cortex.go

```diff
@@ -0,0 +1,226 @@
+	RemoteTimeout time.Duration
+	ProxyURL      string
+	client        *http.Client
+}
+
+type CortexMetricSinkConfig struct {
+	URL           string        `yaml:"url"`
+	RemoteTimeout time.Duration `yaml:"remote_timeout"`
+	ProxyURL      string        `yaml:"proxy_url"`
+}
+
+// Create creates a new Cortex sink.
+// @see veneur.MetricSinkTypes
+func Create(
+	server *veneur.Server, name string, logger *logrus.Entry,
+	config veneur.Config, sinkConfig veneur.MetricSinkConfig,
+) (sinks.MetricSink, error) {
+	conf, ok := sinkConfig.(CortexMetricSinkConfig)
+	if !ok {
+		return nil, errors.New("invalid sink config type")
+	}
+
+	return NewCortexMetricSink(conf.URL, conf.RemoteTimeout, conf.ProxyURL)
+}
+
+// ParseConfig extracts Cortex specific fields from the global veneur config
+func ParseConfig(config interface{}) (veneur.MetricSinkConfig, error) {
+	cortexConfig := CortexMetricSinkConfig{}
+	err := util.DecodeConfig(config, &cortexConfig)
+	if err != nil {
+		return nil, err
+	}
+	if cortexConfig.RemoteTimeout == 0 {
+		cortexConfig.RemoteTimeout = DefaultRemoteTimeout
+	}
+	return cortexConfig, nil
+}
+
+// NewCortexMetricSink creates and returns a new instance of the sink
+func NewCortexMetricSink(URL string, timeout time.Duration, proxyURL string) (*CortexMetricSink, error) {
+	return &CortexMetricSink{
+		URL:           URL,
+		RemoteTimeout: timeout,
+		ProxyURL:      proxyURL,
+	}, nil
+}
+
+// Name returns the string cortex
+func (s *CortexMetricSink) Name() string {
+	return "cortex"
+}
+
+// Start sets up the HTTP client for writing to Cortex
+func (s *CortexMetricSink) Start(*trace.Client) error {
+	// Default concurrent connections is 2
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	t.MaxIdleConns = MaxConns
+	t.MaxConnsPerHost = MaxConns
+	t.MaxIdleConnsPerHost = MaxConns
+
+	// Configure proxy URL, if supplied
+	if len(s.ProxyURL) > 0 {
+		p, err := url.Parse(s.ProxyURL)
+		if err != nil {
+			return errors.Wrap(err, "malformed cortex_proxy_url")
+		}
+		t.Proxy = http.ProxyURL(p)
+	}
+
+	s.client = &http.Client{
+		Timeout:   time.Duration(s.RemoteTimeout * time.Second),
+		Transport: t,
+	}
+	return nil
+}
+
+// Flush sends a batch of metrics to the configured remote-write endpoint
+func (s *CortexMetricSink) Flush(ctx context.Context, metrics []samplers.InterMetric) error {
+	wr := makeWriteRequest(metrics)
+	data, err := proto.Marshal(wr)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	encoded := snappy.Encode(nil, data)
+	buf.Write(encoded)
+
+	req, err := http.NewRequest("POST", s.URL, &buf)
+	if err != nil {
+		return err
+	}
+
+	// This set of headers is prescribed by the remote-write standard
+	req.Header.Set("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("User-Agent", "veneur/cortex")
+	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+
+	r, err := s.client.Do(req)
+	// Returning err here would be neater, but fails go vet
```

Comment:
You do seem to be returning `err` here?

Reply:
Yeah, I think this comment doesn't add much, I'll prune it. FWIW, I meant instead of 

```
if err != nil {
return err
}

return nil
```
one could simply
```
return err
```

But this upsets go vet, which wants to see an error check before the body is accessed.